### PR TITLE
MTurk tasks for the LIGHT project

### PR DESCRIPTION
**Pull request by JackUrb**
Adds the ParlAI MTurk tasks from the LIGHT project. This includes the `light_chats` task, which was used to collect the dataset of conversations, as well as `light_chat_evals`, which was used to test human performance on the retrieval version of the task. 

It also includes making the `light-environment.pkl` file available, which has all of the locations, characters, and objects that we used in the paper.

Lastly, it dumps a slightly cleaned and commented file for version 3 of the graph engine, with fair warnings that it is specifically associated with this task and future work in the area may work on newer versions as we further iterate on the graph engine.

### CODE REVIEW

File: parlai/tasks/light_dialog/build.py [hunk 1]
```diff
@@ -28,9 +28,21 @@ def build(opt):
         url = 'https://dl.fbaipublicfiles.com/parlai/light/light-unseen-processed2.pkl'
         fname = 'light_unseen_data.pkl'
         build_data.download(url, dpath, fname)
+
+        # Download the environment dataset.
+        url = 'https://dl.fbaipublicfiles.com/parlai/light/light-environment.pkl'
```
Comment (emilydinan) [inline parlai/tasks/light_dialog/build.py]: you can use http://parl.ai/downloads as an alias for https://dl.fbaipublicfiles.com/parlai/

**Comment by JackUrb** [comment]
For the failed datatest:
```
jju-mbp1:ParlAI jju$ python tests/datatests/test_new_tasks.py
.
----------------------------------------------------------------------
Ran 1 test in 29.742s

OK
```

File: parlai/mturk/tasks/light/light_chat_eval/run.py [hunk 2]
```diff
@@ -0,0 +1,249 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+from parlai.core.params import ParlaiParser
+from parlai.mturk.tasks.light.light_chat_eval.worlds import \
```
Comment (stephenroller) [inline parlai/mturk/tasks/light/light_chat_eval/run.py]: no \

File: parlai/mturk/tasks/light/light_chat_eval/task_config.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+and everywhere the HIT is mentioned.
+"""
+task_config['hit_title'] = 'Play a character in a room of a text adventure game'
+
+
+"""A description includes detailed information about the kind of task the HIT contains.
+On the Amazon Mechanical Turk web site, the HIT description appears in the expanded
+view of search results, and in the HIT and assignment screens.
+"""
+task_config['hit_description'] = \
```
Comment (stephenroller) [inline parlai/mturk/tasks/light/light_chat_eval/task_config.py]: no \

File: parlai/mturk/tasks/light/light_chat_eval/worlds.py [hunk 4]
```diff
@@ -0,0 +1,372 @@
...
+        )
+        second_act = {
+            'id': 'System',
+            'text': 'SECOND_TURN',
+            'task_data': {
+                'wrong': 0,
+                'turn': 'SECOND_TURN',
+                'curr_message_context': {'action': 'gesture frown'},
+                'actions': [
+                    'You think you can say whatever you want '
```
Comment (stephenroller) [inline parlai/mturk/tasks/light/light_chat_eval/worlds.py]: the mix of newlines with and without , is rather jarring. Should we just override line length?

**Pull request merged**